include/net/lwm2m_client_utils.h

```diff
@@ -25,12 +25,36 @@
 
 #include <zephyr.h>
 #include <net/lwm2m.h>
+#include <modem/lte_lc.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 #if defined(CONFIG_LWM2M_CLIENT_UTILS_SECURITY_OBJ_SUPPORT)
+/**
+ * @typedef modem_mode_cb_t
+ * @brief Callback to request a modem state change, being it powering off, flight mode etc.
+ *
+ * @return 0 if mode was set successfully
+ * @return positive value to indicate seconds before retry
+ * @return -EINVAL if input argument was invalid.
+ * @return -EFAULT if the functional mode could not be retrieved from the modem.
```

Comment:
Should we allow any other errors as well?
Maybe the application just don't want to switch the mode. `EBUSY`?